handlers/common.go

```diff
@@ -130,27 +128,214 @@ func GetValidatorEarnings(validators []uint64, currency string) (*types.Validato
 		return nil, nil, err
 	}
 
-	apr = (((float64(c.EarningsLastWeek) / 1e9) / (float64(32))) * 365) / 7
-	if apr < float64(-1) {
-		apr = float64(-1)
+	// calculate combined el and cl earnings
+	earnings1d := ((income.ClIncome1d * 1e9) + income.ElIncome1d) / 1e9
+	earnings7d := ((income.ClIncome7d * 1e9) + income.ElIncome7d) / 1e9
+	earnings31d := ((income.ClIncome31d * 1e9) + income.ElIncome31d) / 1e9
```

Comment:
you're right, I assumed that I would need to calculate everything in Wei before converting to Gwei in order to avoid precision loss, but after thinking about it, it makes no difference.

adressed in new commit, ptal